### PR TITLE
6879: Remove floppy save icon 

### PR DESCRIPTION
**Pull request by MuhweziDeo**
## Description

This introduces an option to hide save Icon in TocItemSettings 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Enhancement

## Issue

**What is the current behavior?**
Save button is shown
#6879

**What is the new behavior?**
Remove the save button 
## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by tdipisa** [comment]
@MuhweziDeo I saw this in description:

**Comment by offtherailz** [comment]
Moreover @MuhweziDeo I think we should remove also the prompt shown when you edit something, than click on "X" that asks you to confirm save. 
image

Agree @tdipisa ?

**Comment by MuhweziDeo** [comment]
> @MuhweziDeo I saw this in description:
> 
> > This introduces an option to hide save Icon in TocItemSettings
> 
> It is not needed here to introduce a config option to hide the button. It need to be completely removed by making transparent the update of the app state when the user change something in layer settings. See my comment [here](https://github.com/geosolutions-it/mapstore2/issues/6879#issuecomment-844000343) in the issue.

My bad got you now

**Comment by offtherailz** [comment]
So summarising:
 - Remove the save button
 - Remove the `tocItemsSettings` enhancer that takes care of saving the old values to restore them (incuded the parts about the confirm modal on close), and remap methods if needed.

Ok @MuhweziDeo ?

**Comment by MuhweziDeo** [comment]
> So summarising:
> 
> * [ ]  Remove the save button
> * [ ]  Remove the `tocItemsSettings` enhancer that takes care of saving the old values to restore them (incuded the parts about the confirm modal on close), and remap methods if needed.
> 
> Ok @MuhweziDeo ?

ok let me update my implementation

File: web/client/components/TOC/enhancers/tocItemsSettings.js [hunk 1]
```diff
@@ -35,31 +34,7 @@ export const settingsState = compose(
...
-                );
-                onHideSettings();
-                onShowAlertModal(false);
-                // clean up internal settings state
-                onUpdateOriginalSettings({});
-                onUpdateInitialSettings({});
-            }
-        },
-        onSave: ({ onUpdateInitialSettings = () => {}, onUpdateOriginalSettings = () => {}, onHideSettings = () => { }, onShowAlertModal = () => { } }) => (tabsCloseActions = []) => {
+        onClose: ({ onUpdateInitialSettings = () => {}, onHideSettings = () => {} }) => (forceClose, tabsCloseActions = []) => {
```
Comment (dsuren1) [inline web/client/components/TOC/enhancers/tocItemsSettings.js:37]:
```suggestion
        onClose: ({ onUpdateInitialSettings = () => {}, onHideSettings = () => {} }) => (tabsCloseActions = []) => {
```
I think `forceClose` is not required anymore. Also kindly make changes in TOCItemSettings `onClose` call and corresponding unit tests in `tocItemSettings`

File: web/client/components/TOC/enhancers/tocItemsSettings.js [hunk 2]
```diff
@@ -68,9 +43,7 @@ export const settingsLifecycle = compose(
                 });
             }
             onHideSettings();
-            onShowAlertModal(false);
             // clean up internal settings state
-            onUpdateOriginalSettings({});
```
Comment (dsuren1) [inline web/client/components/TOC/enhancers/tocItemsSettings.js:73]: I think `onUpdateOriginalSettings` should be called to reset on `onClose` too

**Review by dsuren1 — CHANGES_REQUESTED** [review]
Looks fine to me. However I have few minor changes requested as above. Kindly see to it

**Review by dsuren1 — APPROVED** [review]
LGTM

**Comment by dsuren1** [comment]
Hi @ElenaGallo. This issue is ready for testing in DEV

**Pull request merged**